.github/workflows/build.yml

```diff
@@ -308,3 +310,194 @@ jobs:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/download-artifact@v2
+      with:
+        name: wheels
+        path: dist/
+
+    - run: |
+        zip -r dist.zip dist/
+
+    - uses: actions/github-script@v5
+      id: asset
+      with:
+        github-token: ${{ secrets.gh_access_token }}
+        script: |
+          const fs = require('fs');
+
+          // Get the ref for HEAD [A call to the API is probably useless]
+          let ref_head = await github.rest.git.getRef({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            ref: 'heads/master',
+          });
+          console.log(`HEAD reference ${ref_head.data.object.sha}`);
+
+          // Retrieve ref for tag `nightly`
+          let ref_nightly = null;
+          try {
+            ref_nightly = await github.rest.git.getRef({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              ref: 'tags/nightly',
+            });
+
+            if (ref_nightly.data.object.sha === ref_head.data.object.sha) {
+              console.log('No new nightly release');
+              return;
+            }
+          } catch (err) {
+            // The tag does not exist so let's create it
+            ref_nightly = await github.rest.git.createRef({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              ref: 'refs/tags/nightly',
+              sha: ref_head.data.object.sha,
+            });
+          }
+
+          // Call the GitHub API to get a release by tag
+          let release = null;
+          try {
+            release = await github.rest.repos.getReleaseByTag({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              tag: 'nightly',
+            });
+            console.log(`Found release ${release.data.tag_name} ${release.data.draft} ${release.data.prerelease}`);
+          } catch (err) {
+            console.log(`Release ${tag} not found`);
+
+            // If the release doesn't exist, create it
+            release = await github.rest.repos.createRelease({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              tag_name: 'nightly',
+              name: 'nightly',
+              body: 'Nightly release crafted with ♥️ somewhere on 🌎',
+              draft: false,
+              prerelease: true,
+            });
+
+            console.log(`Created release ${release.data.tag_name} ${release.data.draft} ${release.data.prerelease} ${doIProceed}`);
+          }
+          console.log(`Release does exist with tag ${release.data.tag_name} [${release.data.draft} ${release.data.prerelease}]`);
+
+          // At this stage both tag & release exist
+
+          // Update nightly tag
+          await github.rest.git.updateRef({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            ref: 'tags/nightly',
+            sha: ref_head.data.object.sha,
```

Comment:
It would be wise to add `force: true` in case master has been rebased non fast-forward.

Reply:
done